### ISSUE TITLE
`decompose_essential_matrix_no_svd` returns non-rotations for any batch of two or more, even `[E, E]`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4880]
`decompose_essential_matrix_no_svd(E)` implements Horn's closed form, `R = (matrix_cofactor_tensor(E) − [b]ₓ E) / (b · b)`. The normaliser `(b1 * b1).sum()` (`essential.py:605-606`, and `b2` alike) sums over the whole batch instead of per matrix, so every rotation is divided by the sum of all elements' `b · b`; for a batch of B copies of one matrix that is B times too much: for `[E, E]` each returned `R` is exactly half the correct one (`‖R Rᵀ − I‖ = 1.3`, `det = 0.125`), for `[E, E, E]` a third. `t` is unaffected, and a batch of one is correct. The docstring promises rotations for any `(*, 3, 3)` input.

No test sees it: `TestDecomposeEssentialMatrixNoSVD.test_shape` checks only shapes for batches larger than one, and `test_correct_decompose`, `test_gradcheck` and `test_consistency` use a batch of one. `test_consistency` in `tests/geometry/epipolar/test_essential.py` carries a non-strict `xfail` ("skip the tests where there are no solutions."), so the suite asserts nothing about its outcome; it XPASSes today on CPU float32 and float64.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import decompose_essential_matrix_no_svd, essential_from_Rt

R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]]))
t = torch.tensor([[[0.5], [0.1], [0.02]]])
E = essential_from_Rt(torch.eye(3)[None], torch.zeros(1, 3, 1), R, t)   # (1, 3, 3)
I = torch.eye(3)

def report(name, Eb):
    R1, R2, tt = decompose_essential_matrix_no_svd(Eb)
    for i in range(Eb.shape[0]):
        orth = max((R1[i] @ R1[i].T - I).norm().item(), (R2[i] @ R2[i].T - I).norm().item())
        print(f"{name:9s} element {i}: max ||R R^T - I|| = {orth:.3g}, det(R1) = {torch.linalg.det(R1[i]).item():.4g}")

report("[E]", E)
report("[E, E]", torch.cat([E, E]))
report("[E, E, E]", torch.cat([E, E, E]))
R1s = decompose_essential_matrix_no_svd(E)[0]
for B in (2, 3):
    R1b = decompose_essential_matrix_no_svd(E.expand(B, 3, 3))[0]
    print(f"B = {B}: max|B * R1_batched - R1_single| = {(B * R1b - R1s).abs().max().item():.2g}")
```

Output:

```text
[E]       element 0: max ||R R^T - I|| = 2.8e-07, det(R1) = 1
[E, E]    element 0: max ||R R^T - I|| = 1.3, det(R1) = 0.125
[E, E]    element 1: max ||R R^T - I|| = 1.3, det(R1) = 0.125
[E, E, E] element 0: max ||R R^T - I|| = 1.54, det(R1) = 0.03704
[E, E, E] element 1: max ||R R^T - I|| = 1.54, det(R1) = 0.03704
[E, E, E] element 2: max ||R R^T - I|| = 1.54, det(R1) = 0.03704
B = 2: max|B * R1_batched - R1_single| = 0
B = 3: max|B * R1_batched - R1_single| = 6e-08
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same), CPU float32, macOS arm64.

## Classification

Clearly broken output. No idiom of 1–9; candidate: a reduction over the batch axis where a per-element reduction is meant.

## Disposition

Focused fix, welcome as a PR: reduce per matrix, `(b1 * b1).sum(-1)[:, None, None]` (and `b2`), and add a batched value test; the non-strict `xfail` on `test_consistency` can go, since it passes. A batch of one is unchanged.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
